apps/challenges/views.py

```diff
@@ -2501,6 +2532,55 @@ def get_or_update_challenge_phase_split(request, challenge_phase_split_pk):
         return Response(response_data, status=status.HTTP_200_OK)
 
 
+@api_view(["PATCH"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticatedOrReadOnly, HasVerifiedEmail))
+@authentication_classes((JWTAuthentication, ExpiringTokenAuthentication))
+def update_challenge_tags_and_domain(request, challenge_pk):
+    """
+    Returns or Updates challenge tags and domain
+    """
+    challenge = get_challenge_model(challenge_pk)
+
+    if request.method == "PATCH":
+        new_tags = request.data.get("list_tags", [])
+        domain_value = request.data.get("domain")
+        # Remove tags not present in the YAML file
+        challenge.list_tags = [tag for tag in challenge.list_tags if tag in new_tags]
+        # Add new tags to the challenge
+        for tag_name in new_tags:
+            if tag_name not in challenge.list_tags:
```

Comment:
why same ifs?

Reply:
Sorry that is a mistake of wrong conflict merge by me. i will remove it